frontend/aml_dashboard/src/index.js

```diff
@@ -0,0 +1,746 @@
+const instanceViewer = imageDisplay(vis_detection);
+
+hand_window
+  .filter(() => capture.$pressed.get())
+  .map(async (arr) => (calculate_time_features(arr)))
+  .awaitPromises()
+  .subscribe(trainingSet.create);
+
+
+// -----------------------------------------------------------
+// TRAINING
+// -----------------------------------------------------------
+
+const b = button('Train');
+b.title = 'Neural Network Training Launcher';
+const classifier = mlpClassifier({ layers: [64, 32], epochs: 20 }).sync(store, 'mlp-dashboard');
+
+b.$click.subscribe(() => classifier.train(trainingSet));
+
+const url_train = "http://localhost:5000/train/";
+const aml_model = { "ready" : false};
+
+const b_train_AML = button('Train');
+b_train_AML.title = 'AML Training Launcher'
+
+const textAMLTrainStatus = text('Not Trained');
+textAMLTrainStatus.title = 'AML Status';
+
+const n_jobs = number(1);
+n_jobs.title = 'Parallel Training (executions)';
+
+const features = {
+  parameters: {
+    'Iterations': new Stream(10, true),
+    'Percentage of data': new Stream(20, true),
+  },
+};
+
+const controls = modelParameters(features);
+controls.title = 'Choose input values per execution';
+
+console.log('Controls:', controls);
+
+b_train_AML.$click.subscribe( async () => {
+  textAMLTrainStatus.$value.set(' <h1>Training...</h1> <br> <img src="https://i.gifer.com/origin/05/05bd96100762b05b616fb2a6e5c223b4_w200.gif">');
+
+  const v = await trainingSet.find();
+
+  // TODO: save the training set to a file each time is updated
+  save_to_file(v, "training_set_.json"); // Save the training set to a file
+
+  const json_dt = {x : [], y : []};
+  const dt = v.data;
+  for(let i = 0; i < dt.length; i++) {
+    json_dt.x.push(dt[i].x);
+    json_dt.y.push(dt[i].y);
+  }
+
+  const n_job = n_jobs.$value.get().toString();
+  const n_iter = features.parameters['Iterations'].value;
+  const percentage_data = features.parameters['Percentage of data'].value;
+
+  fetch(url_train + n_job + '/' + n_iter + '/' + percentage_data, {
+    method: "POST", // *GET, POST, PUT, DELETE, etc.
+    mode: "cors", // no-cors, *cors, same-origin
+    cache: "no-cache", // *default, no-cache, reload, force-cache, only-if-cached
+    credentials: "omit", // include, *same-origin, omit
+    headers: {
+      "Content-Type": "application/json",
+    },
+    redirect: "follow", // manual, *follow, error
+    referrerPolicy: "no-referrer", // no-referrer, *no-referrer-when-downgrade, origin, origin-when-cross-origin, same-origin, strict-origin, strict-origin-when-cross-origin, unsafe-url
+    body: JSON.stringify(json_dt), // body data type must match "Content-Type" header
+  })
+  .then(response => response.json())
+  .then(async json => {
+    console.log(json);
+    console.log('RECEIVED');
+
+    save_to_file(json, "model_.json");
+
+    aml_model['ready'] = true;
+
+    textAMLTrainStatus.$value.set('<h1>Finished :)</h1> ');
```

Comment:
Please add a case for when there are errors or the training could not be finished.
```suggestion
    textAMLTrainStatus.$value.set('<h1>Therehas been an error. Try again! :( </h1> ');
```

Reply:
Same as the previous one.

Reply:
Done

Reply:
Addressed in #2